cpp/CMakeLists.txt

```diff
@@ -141,6 +145,22 @@ if(NOT BUILD_CUML_CPP_LIBRARY)
   set(BUILD_CUML_EXAMPLES OFF)
 endif(NOT BUILD_CUML_CPP_LIBRARY)
 
+# SingleGPU build disables cumlprims_mg and comms components
+if(SINGLEGPU)
+  message(STATUS "Detected SINGLEGPU build option")
+  message(STATUS "Disabling Multi-GPU components and comms libraries")
+  set(BUILD_CUML_MG_TESTS OFF)
+  set(BUILD_CUML_STD_COMMS OFF)
+  set(BUILD_CUML_MPI_COMMS OFF)
+  set(DISABLE_CUMLPRIMS_MG ON)
```

Comment:
Nice that this disables all... but why is everything else and ENABLE flag and this one is DISABLE? Maybe confusing to deal with both senses? Or is that necessary to support the default?